browser-extension/mv2/src/background/background.js

```diff
@@ -1069,6 +1074,20 @@ BG.Methods.onContentScriptLoadedNotification = async (tabId) => {
   }
 };
 
+BG.Methods.startRecordingOnUrlListener = (tabId, changeInfo, shouldListen) => {
+  if (shouldListen && changeInfo.status === "complete") {
+    chrome.tabs.sendMessage(tabId, { action: RQ.CLIENT_MESSAGES.START_RECORDING });
+  }
+};
+
+BG.Methods.startRecordingOnUrl = (url) => {
+  chrome.tabs.create({ url }, (tab) => {
+    chrome.tabs.onUpdated.addListener((tabId, changeInfo) =>
```

Comment:
If I manually change the URL after page has loaded, will it start recording on the new URL?

Reply:
Yes, it will as matching is based on tab id

Reply:
Ideally, it should record only for the URL launched from App.